File: lab8/GCD.c
/*
 * GCD.c
 * Juan Ayala
 * jumaayal
 * lab8
 * Finds the GCD of two positive integers
 */
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include<ctype.h>

int isInteger(char s[]){
   int i, n;

   if( s==NULL )
      return 0;
   n = strlen(s);
   if( n==0 )
      return 0;
   if( s[0]!='-' && s[0]!='+' && !isdigit(s[0]) )
      return 0;
   for(i=1; i<n; i++){
      if( !isdigit(s[i]) )
         return 0;
   }
   return 1;
}

int main(void){
	int x, y,n,p,dividend,divisor,remainder =1;
	char str[200];
	printf("Enter a positive integer: ");
	while(1){
		n = scanf("%s", str);
		if( isInteger(str) ){
			sscanf(str, "%d", &x);
			if(x>0){
				break;
			}
		}
		printf("Please enter a positive integer: ");
	}

	printf("Enter another positive integer: ");
	while(1){
		p = scanf("%s", str);
		if( isInteger(str) ){
			sscanf(str, "%d", &y);
			if(x>0){
				break;
			}
		}
		printf("Please enter a positive integer: ");
	}
	if(x>y){
		dividend = x;
		divisor=y;
	}else{
		dividend = y;
		divisor = x;
	}
	while(remainder !=0){
		remainder = dividend%divisor;
		dividend = divisor;
		divisor = remainder;
	}
	printf("The GCD of %d and %d is %d\n", x,y,dividend);
}

File: PA3/GCD.java
//GCD.java
//Juan Ayala
//jumaayal
//pa3
//Finds greatest common divisor
import java.util.Scanner;
class GCD{
	public static void main (String[] args){
		int firstInt;
		int secInt;
		int quotient =0;
		int remainder=1;
		int dividend;
		int divisor;
		
		Scanner sc = new Scanner(System.in);
		
		//Checks whether first input is an integer and positive
		System.out.print("Enter a positive integer: ");
		while(true){
			while((!sc.hasNextInt())){
				System.out.print("Please enter a positive integer: ");
				sc.next();
			}
			firstInt = sc.nextInt();
			if(firstInt > 0){
				break;
			}else{
				System.out.print("Please enter a positive integer: ");
			}
		}
		
		//Checks whether second input is an integer and positive
		System.out.print("Enter another positive integer: ");
		while(true){
			while((!sc.hasNextInt())){
				System.out.print("Please enter a positive integer: ");
				sc.next();
			}
			secInt = sc.nextInt();
			if(secInt > 0){
				break;
			}else{
				System.out.print("Please enter a positive integer: ");
			}
		}
			
		//Figures which input is larger and assign appropriate names	
		if(secInt>firstInt){
			dividend = secInt;
			divisor= firstInt;
		}else{
			dividend = firstInt;
			divisor = secInt;
		}
				 
		//Loops through Euclidean Algorithm to find quotient
		while(remainder!=0){
			remainder = dividend%divisor;
			dividend = divisor;
			divisor = remainder;
		}
		
		
		System.out.println("The GCD of " + firstInt+ " and " + secInt  + " is " + dividend);
	}
}
